### PR TITLE
Added Recursive Bubble Sort

### PR DESCRIPTION
**Pull request by AdityaPrakash-26**
I have implemented the recursive version of Bubble Sort as per Issue #97 

Since I could not find implementations for all languages under the organisation, I have included a Pseudocode for convenience.

### CODE REVIEW

File: en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+- We need to fix the last element of the current sub-array. For this, iterate over the entire array using normal Bubble Sort, and perform swapping.
+- Next, call the function on the entire array excluding the last element(which was fixed by the iteration in the above step)
+- Repeat until Base Case is reached.
+
+## Example:
+
+Let the given array be: `{5, 3, 2, 1, 4}`
+
+__First Pass:__
```
Comment (Panquesito7) [inline en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md:20]: Shouldn't `pass` be `test`?

File: en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    for(i = 0; i<n-1; i++)
+        if(arr[i] > arr[i+1])
+            swap(arr[i], arr[i+1])
+
+    bubbleSort(arr, n-1)
+```
+
+## Implementations:
+-[Python](https://github.com/TheAlgorithms/Python/blob/master/searches/binary_search.py)
```
Comment (Panquesito7) [inline en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md:66]: Is this binary search or bubble sort? If it's bubble sort, you should link to a bubble sort implementation.
```suggestion
- [Python](https://github.com/TheAlgorithms/Python/blob/master/sorts/bubble_sort.py)
```

**Review by Panquesito7 — COMMENTED** [review]
👍 Good work.

File: en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md (hunk 1 above)
Comment (AdityaPrakash-26) [reply en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md:20]: Hrmmm. I think First `Iteration` would be more proper? By pass, I meant the number of times I have traversed the array.

File: en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md (hunk 2 above)
Comment (AdityaPrakash-26) [reply en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md:66]: Oops! My bad. I got mixed up between searches and sorting. I just had a look at bubble_sort.py in the repo, but they don't seem to have a recursive version of the algorithm.

Comment (Panquesito7) [reply en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md:66]: You can also add other implementations in other languages if they exist. 🙂

Comment (AdityaPrakash-26) [reply en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md]: I found implementations in C and Java. I have added those

File: en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md (hunk 1 above)
Comment (Panquesito7) [reply en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md:20]: OK, then use `iteration` instead. 🙂

File: en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    for(i = 0; i<n-1; i++)
+        if(arr[i] > arr[i+1])
+            swap(arr[i], arr[i+1])
+
+    bubbleSort(arr, n-1)
+```
+
+## Implementations:
+-[Java](https://github.com/TheAlgorithms/Java/blob/master/Sorts/BubbleSortRecursion.java)
+-[C](https://github.com/TheAlgorithms/C/blob/master/sorting/bubble_sort_recursion.c)
```
Comment (Panquesito7) [inline en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md:67]:
```suggestion
- [Java](https://github.com/TheAlgorithms/Java/blob/master/Sorts/BubbleSortRecursion.java)
- [C](https://github.com/TheAlgorithms/C/blob/master/sorting/bubble_sort_recursion.c)
```

File: en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+## Video Explanation:
+
+[A video explaining iterative as well as recursive bubble sort](https://www.youtube.com/watch?v=gDMDVLBfCI0)
+
+
```
Comment (Panquesito7) [inline en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md:73]:
```suggestion
```

File: en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md (hunk 1 above)
Comment (AdityaPrakash-26) [reply en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md]: Thanks! Fixed in latest commit

File: en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+- {`1`, `2`, 3, 4, 5}; As `1 < 2`, do not swap
+
+__Fifth Iteration:__
+
+- {`1`, 2, 3, 4, 5}; As the size of the array is 1, return.
+
+Note: This is the base case.
+
+## Pseudo Code:
```
Comment (Panquesito7) [inline en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md:52]:
```suggestion
## Pseudo Code

```

File: en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        return;
+
+    for(i = 0; i<n-1; i++)
+        if(arr[i] > arr[i+1])
+            swap(arr[i], arr[i+1])
+
+    bubbleSort(arr, n-1)
+```
+
+## Implementations:
```
Comment (Panquesito7) [inline en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md:65]:
```suggestion
## Implementations

```

File: en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+- [Java](https://github.com/TheAlgorithms/Java/blob/master/Sorts/BubbleSortRecursion.java)
+- [C](https://github.com/TheAlgorithms/C/blob/master/sorting/bubble_sort_recursion.c)
+
+## Video Explanation:
```
Comment (Panquesito7) [inline en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md:69]:
```suggestion
## Video Explanation
```

File: en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+## Steps
+
+Base case: If the size of the array is 1, return.
+
+- We need to fix the last element of the current sub-array. For this, iterate over the entire array using normal Bubble Sort, and perform swapping.
+- Next, call the function on the entire array excluding the last element(which was fixed by the iteration in the above step)
+- Repeat until Base Case is reached.
+
+## Example:
```
Comment (Panquesito7) [inline en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md:16]:
```suggestion
## Example
```

File: en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+- We need to fix the last element of the current sub-array. For this, iterate over the entire array using normal Bubble Sort, and perform swapping.
+- Next, call the function on the entire array excluding the last element(which was fixed by the iteration in the above step)
+- Repeat until Base Case is reached.
+
+## Example
+
+Let the given array be: `{5, 3, 2, 1, 4}`
+
+__First Iteration:__
```
Comment (Panquesito7) [inline en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md:20]:
```suggestion
**First Iteration:**
```

File: en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+
+- {`5`, `3`, 2, 1, 4} -> {`3`, `5`, 2, 1, 4} Swap since `5 > 3`
+- {3, `5`, `2`, 1, 4} -> {3, `2`, `5`, 1, 4} Swap since `5 > 2`
+- {3, 2, `5`, `1`, 4} -> {3, 2, `1`, `5`, 4} Swap since `5 > 1`
+- {3, 2, 1, `5`, `4`} -> {3, 2, 1, `4`, `5`} Swap since `5 > 4`
+
+This iteration has fixed the position of 5. Now, we will consider the array up to index 3.
+
+__Second Iteration:__
```
Comment (Panquesito7) [inline en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md:29]:
```suggestion
**Second Iteration:**
```

File: en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+
+- {`3`, `2`, 1, 4, 5} -> {`2`, `3`, 1, 4, 5} Swap since `3 > 2`
+- {2, `3`, `1`, 4, 5} -> {2, `1`, `3`, 4, 5} Swap since `3 > 1`
+- {2, 1, `3`, `4`, 5}; As `3 < 4`, do not swap
+
+Note: As we check one less element with every iteration, we do not need elements at index 3 and 4 i.e., `4` and `5`, as 5 is already in order. Formally, for an array with `n` integers, we consider elements only up to index `n - i`, where `i` is the iteration number.
+
+__Third Iteration:__
```
Comment (Panquesito7) [inline en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md:37]:
```suggestion
**Third Iteration:**
```

File: en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+
+- {`2`, `1`, 3, 4, 5} -> {`1`, `2`, 3, 4, 5} Swap since `1 > 2`
+- {1, `2`, `3`, 4, 5}; As `2 < 3`, do not swap
+
+__Fourth Iteration:__
```
Comment (Panquesito7) [inline en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md:42]:
```suggestion
**Fourth Iteration:**
```

File: en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+
+- {`1`, `2`, 3, 4, 5}; As `1 < 2`, do not swap
+
+__Fifth Iteration:__
```
Comment (Panquesito7) [inline en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md:46]:
```suggestion
**Fifth Iteration:**
```

File: en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md [hunk 14]
```diff
@@ -0,0 +1,73 @@
+# Recursive Bubble Sort
+
+Bubble Sort is one of the simplest sorting algorithms that compares two elements at a time and swaps them if they are in the wrong order. This process is repeated until the entire sequence is in order.
+
+- Time Complexity: `O(n ^ 2)` for average case; `O(n)` for best case
+- Space Complexity: `O(n)`; note that iterative bubble sort has space complexity as `O(1)`
```
Comment (Panquesito7) [inline en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md:6]:
```suggestion
- Time Complexity: `O(n ^ 2)` for average case; `O(n)` for best case.
- Space Complexity: `O(n)`; note that iterative bubble sort has space complexity as `O(1)`.
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Review by Panquesito7 — APPROVED** [review]
LGTM. Thank you for your contribution! 👍 🎉

**Comment by AdityaPrakash-26** [comment]
Thanks a lot! Looking forward to making more pull requests :-)

**Review by vbrazo — APPROVED** [review]
LGTM!

**Comment by vbrazo** [comment]
Looks awesome 💯

**Pull request merged**